mllib/src/main/scala/org/apache/spark/mllib/clustering/BisectingKMeans.scala

```diff
@@ -194,8 +196,9 @@ class BisectingKMeans private (
           newClusters = summarize(d, newAssignments)
           newClusterCenters = newClusters.mapValues(_.center).map(identity)
         }
-        // TODO: Unpersist old indices.
-        val indices = updateAssignments(assignments, divisibleIndices, newClusterCenters).keys
+        if (preIndices != null) preIndices.unpersist()
+        preIndices = indices
+        indices = updateAssignments(assignments, divisibleIndices, newClusterCenters).keys
```

Comment:
You are probably ahead of me on this, but let me check something. To compute `assignments` in the next line, the current `indices` is needed, and that in turn needs the current value of `assignments`, which needs the previous copy of `indices` (`preIndices`). But that was unpersisted just above. Should `preIndices` be unpersisted later, after `indices` is materialized?

But ... is there even an action here? if this just creating a large lineage then the persisting isn't helping much.